pwem/constants.py

```diff
@@ -69,59 +71,72 @@
 COLOR_GNU_PLOT2 = 5
 COLOR_OTHER = 6
 
-COLOR_CHOICES = ['jet', 'terrain',  'gist_earth', 'gist_ncar', 'gnuplot', 'gnuplot2', 'other']
+COLOR_CHOICES = ['jet', 'terrain', 'gist_earth',
+                 'gist_ncar', 'gnuplot', 'gnuplot2', 'other']
 
 # Axis code
 AX_X = 0
 AX_Y = 1
 AX_Z = 2
 
-
-# SYMMETRY, conventions described atr: https://scipion-em.github.io/docs/docs/developer/symmetries
+# SYMMETRY, conventions described at:
+# https://scipion-em.github.io/docs/docs/developer/symmetries
 # Some notes:
 # Icosahedral
 #   xmipp and relion define I1, I2, I3 and I4 that corerspond to
 #   I222, I222r, In25 and In25z
-#   cryosparc has I1 and I2 
-#   EMAN always puts the highest symmetry axis on Z. 
+#   cryosparc has I1 and I2
+#   EMAN always puts the highest symmetry axis on Z.
 #   and a 2 fold axis in X (so it uses I2n5 or I2n5r, not sure)
 # DIEDRAL: first axis Z, second X (DX) except cryosparc that uses y (DY)
-# Tetraedral, most of the programs use T222, EMAN uses Tz3
-SYM_CYCLIC = 0  # CN cyclic symmetry Cn around z axis
-SYM_DIHEDRAL = 1  # DN cyclic symmetry plus and extra 2fold symmetry axis around X
+# Tetraedral, most of the programs use TZ3
+# CN cyclic symmetry Cn around z axis
+SYM_CYCLIC = 0
+# DN cyclic symmetry plus and extra 2fold symmetry axis around X
+SYM_DIHEDRAL = 1
 SYM_DIHEDRAL_X = SYM_DIHEDRAL
-SYM_DIHEDRAL_Y = 2  # DN cyclic symmetry plus and extra 2fold symmetry axis around Y (cryoSparc)
-SYM_TETRAHEDRAL = 3  # T 222 Tetrahedral symmetry with two-fold symmetry axes along the X, Y,
-# and Z axes, a three-fold along axis (1,1,1)
-SYM_TETRAHEDRAL_Z3 = 4  # T  three-fold symmetry axis along Z, another three-fold axis 
-# in the YZ plane such that rotation about the X axis by ~110°
-# is a symmetry operation (EMAN convention)
-
-SYM_OCTAHEDRAL = 5  # O
+# DN cyclic symmetry plus and extra 2fold symmetry axis around Y (cryoSparc)
+SYM_DIHEDRAL_Y = 2
+# T 222 Tetrahedral symmetry with two-fold symmetry axis along the X, Y and Z
+SYM_TETRAHEDRAL = 3
+SYM_TETRAHEDRAL_222 = SYM_TETRAHEDRAL
+# T  three-fold symmetry axis along Z, another three-fold axis
+# in the YZ plane such that the sign(y) != sign(z)
+SYM_TETRAHEDRAL_Z3 = 4
+# SYM_TETRAHEDRAL222R  is a degenerated symmetry, that is
+# for each symmetry axis (x,y,z)  defined in T222, there is a
+# symmetry axis (-x,-y,-z) in T222R, therefore the produced
+# symmetry matrices are equivalent
+# SYM_TETRAHEDRAL222R = 14
+
+# T  three-fold symmetry axis along Z, another three-fold axis
+# in the YZ plane such that  sign(y) = sign(z)
+SYM_TETRAHEDRAL_Z3R = 5
+
+SYM_OCTAHEDRAL = 6  # O
```

Comment:
Octahedral has now 6 instead of 5... would this be a problem? for already existing protocols/workflows?

Reply:
Good question @pconesa . 

  Obviously if someone has hardcoded 5 instead of using SYM_OCTAHEDRAL that will be a problem. I have checked xmipp and crysparc plugins (I think this are the only ones that refer to scipion symmetry and I am not the maintainer). In both cases they first import  SCIPION keys

from pwem.constants import SCIPION_SYM_NAME
from pwem.constants import (
    SYM_CYCLIC, SYM_TETRAHEDRAL, SYM_OCTAHEDRAL, SYM_I222,
    SYM_I222r)

and then define labels for the implemented symmetries:

CS_SYM_NAME = dict()
CS_SYM_NAME[SYM_CYCLIC] = 'Cn'
CS_SYM_NAME[SYM_DIHEDRAL_Y] = 'Dn'
CS_SYM_NAME[SYM_TETRAHEDRAL] = 'T'
CS_SYM_NAME[SYM_OCTAHEDRAL] = 'O'
CS_SYM_NAME[SYM_I222] = 'I1'
CS_SYM_NAME[SYM_I222r] = 'I2'

and finally use them to create choices as:

    form.addParam('symmetryGroup', EnumParam,
                  choices=[CS_SYM_NAME[SYM_CYCLIC] +
                           " (" + SCIPION_SYM_NAME[SYM_CYCLIC] + ")",
                           CS_SYM_NAME[SYM_DIHEDRAL_Y] +
                           " (" + SCIPION_SYM_NAME[SYM_DIHEDRAL_Y] + ")",
                           CS_SYM_NAME[SYM_TETRAHEDRAL] +
                           " (" + SCIPION_SYM_NAME[SYM_TETRAHEDRAL] + ")",
                           CS_SYM_NAME[SYM_OCTAHEDRAL] +
                           " (" + SCIPION_SYM_NAME[SYM_OCTAHEDRAL] + ")",
                           CS_SYM_NAME[SYM_I222] +
                           " (" + SCIPION_SYM_NAME[SYM_I222] + ")",
                           CS_SYM_NAME[SYM_I222r] +
                           " (" + SCIPION_SYM_NAME[SYM_I222r] + ")"],
                  default=SYM_CYCLIC,

this approach seems to be quite robust. Note that the packages have less implemented symmetries that the ones described by scipion so, in general, if they use directly the scipion symmetry definition they can not assume that there are going to get a set of consecutive values.